TourMate/TourMate/Frontend/Plan/Views/PlansViews/PlansListView.swift

```diff
@@ -62,24 +62,53 @@ struct PlansListView: View {
         }
     }
 
-    var body: some View {
+    func createUpvoteView(_ plan: Plan) -> some View {
```

Comment:
We create a viewmodel for upvoting, so when we save upvotes, we don't override the changes made to any plans, just the upvoting.